### PR TITLE
feat: implement personal / favorites folder on the mobile platform

### PR DESCRIPTION
**Pull request by LucasXu0**
### Feature Preview

Screenshot 2023-10-18 at 20 27 32
Screenshot 2023-10-18 at 20 27 43
Screenshot 2023-10-18 at 20 30 40
Screenshot 2023-10-18 at 20 28 36
Screenshot 2023-10-19 at 17 26 53
Screenshot 2023-10-19 at 17 27 31

---

#### PR Checklist

- [x] My code adheres to [AppFlowy's Conventions](https://docs.appflowy.io/docs/documentation/software-contributions/conventions)
- [ ] I've listed at least one issue that this PR fixes in the description above.
- [ ] I've added a test(s) to validate changes in this PR, or this PR only contains semantic changes.
- [ ] All existing tests are passing.

### CODE REVIEW

**Comment by LucasXu0** [comment]
**NOT READY FOR MERGING.**

- [ ] delete
- [ ] rename
- [ ] remove
- [ ] open page

**Comment by LucasXu0** [comment]
> The second item on the bottom navigation bar was planned to show all the favorite pages. I thought we were supposed to have all the personal documents under the home page.

Yes. I should move the favorite section from the home page to the favorite tab.

File: frontend/appflowy_flutter/lib/mobile/presentation/editor/mobile_editor_screen.dart [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+              return const Center(
+                child: CircularProgressIndicator(),
+              );
+            }
+            if (!state.hasData) {
+              // FIXME: handle the error
+              return const Center(
+                child: FlowyText('No data'),
+              );
+            }
```
Comment (Xazin) [inline frontend/appflowy_flutter/lib/mobile/presentation/editor/mobile_editor_screen.dart:53]: Maybe we should have a generic design for a failure screen. I saw we have the "success" screen, and I built this in a minute to illustrate what I mean.

<details>
  <summary>Example of error screen</summary>

image

</details>

File: frontend/appflowy_flutter/lib/mobile/presentation/home/favorite_folder/mobile_home_favorite_folder_header.dart [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        Expanded(
+          child: FlowyButton(
+            text: FlowyText.semibold(
+              LocaleKeys.sideBar_favorites.tr(),
+              fontSize: 20.0,
+            ),
+            margin: const EdgeInsets.symmetric(vertical: 8),
+            expandText: false,
+            mainAxisAlignment: MainAxisAlignment.start,
+            rightIcon: TurnBox(
```
Comment (Xazin) [inline frontend/appflowy_flutter/lib/mobile/presentation/home/favorite_folder/mobile_home_favorite_folder_header.dart:42]: Not sure what this widget is, but this looks like a candidate for https://api.flutter.dev/flutter/widgets/AnimatedRotation-class.html

File: frontend/appflowy_flutter/lib/mobile/presentation/home/mobile_home_page_header.dart [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+                userProfile.email,
+                fontSize: 12,
+                color: Colors.grey,
+              )
+            ],
+          ),
+          const Spacer(),
+          IconButton(
+            icon: const Icon(Icons.more_horiz_outlined),
+            onPressed: () {},
```
Comment (Xazin) [inline frontend/appflowy_flutter/lib/mobile/presentation/home/mobile_home_page_header.dart:44]: Consider adding a TODO

File: frontend/appflowy_flutter/lib/mobile/presentation/home/turn_box.dart [hunk 4]
```diff
@@ -0,0 +1,59 @@
+import 'package:flutter/widgets.dart';
+
+class TurnBox extends StatefulWidget {
```
Comment (Xazin) [inline frontend/appflowy_flutter/lib/mobile/presentation/home/turn_box.dart:3]: AnimatedRotation is just an implicit version of using `RotationTransition`, which means we can avoid having this widget, what do you think?

**Comment by LucasXu0** [comment]
### Error Page
Screenshot 2023-10-23 at 16 20 01

### Calendar
Screenshot 2023-10-23 at 16 46 29

### Board
Screenshot 2023-10-23 at 16 47 12

### Grid
Screenshot 2023-10-23 at 16 47 39

**Pull request merged**